client/src/app/feedback-list/feedback-list.component.spec.ts

```diff
@@ -23,13 +33,55 @@ describe('FeedbackListComponent', () => {
   });
 
   beforeEach(() => {
+    router = TestBed.get(Router);
+    location = TestBed.get(Location);
     fixture = TestBed.createComponent(FeedbackListComponent);
     component = fixture.componentInstance;
-    component.type = FeedbackType.Received
+    component.feedbacks = feedbacks
+    component.type = FeedbackType.Sent
+    router.initialNavigation();
     fixture.detectChanges();
   });
-
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+  it('should show the empty template when feedbacks list is null' ,() => {
+    component.feedbacks = []
+    fixture.detectChanges()
+    const emptyTemplate = fixture.debugElement.query(By.css('.empty'))
+    expect(emptyTemplate).toBeTruthy()
+  });
+  it('empty template should be disapeared when the feedback list is not null', ()=> {
+      const emptyTemplate = fixture.debugElement.query(By.css('.empty'))
+      expect(emptyTemplate).toBeFalsy()
+  });
+  it('the feedback list will be appeared when the list contains feedbacks', ()=> {
+    const feedbackList = fixture.debugElement.query(By.css('.list'))
+    expect(feedbackList).toBeTruthy()
+  });
+  it('should show the sent feedbacks when the feedzbacks envoyés tab is clicked', ()=> {
+    const senderName = fixture.debugElement.query(By.css('a'))
+    expect(senderName.nativeElement.textContent.trim()).toEqual(feedbacks[0].receverName)
+  });
+  it('should show the recevied feedbacks when the feedzbacks reçus tab is clicked', ()=> {
+    component.type = FeedbackType.Received
+    fixture.detectChanges()
+    const senderName = fixture.debugElement.query(By.css('a'))
+    expect(senderName.nativeElement.textContent.trim()).toEqual(feedbacks[0].senderName)
```

Comment:
```suggestion
    expect(senderName.nativeElement.textContent). toContain(feedbacks[0].senderName)
```
Pour éviter de nettoyer le contenu tu peux utiliser toContain